application-ownership-models/README.md

```diff
@@ -0,0 +1,116 @@
+- What stages of the application lifecycle are owned by a team?
+- Is there one or more Giant Swarm teams that can be considered owners of the application?
+
+## 1. Application lifecycle ownership
+
+We can define two high-level application ownership models:
+- Full application ownership
+- Partial application ownership, usually including release engineering (full or partial) and operations. This is typical for upstream applications where the development is done outside of Giant Swarm.
+
+### 1.1. Full application lifecycle ownership
+
+(Almost) Full application ownership includes all stages of the application lifecycle.
+
+Development is done by team members.
+
+Release engineering is done by team members who are using RelEng platform provided by Giant Swarm RelEng teams.
+
+Operations are handled by the team members, including business hours as a part of team on-call, and non-business hours as a part of team/area on-call. There are few caveats here:
+- The non-business hours on-call is currently shared with other teams from the same area (hence "almost" full ownership), which is done for the sake of not overwhelming on-callers with frequent night shifts.
+- Ideally, if a team has enough people and we have state-of-the-art monitoring and alerting, meaning that there are next-to-none false positive alerts (for example caused by alerts from other team's components), the team can own 24h on-call for the application.
+
+### 1.2. Partial application lifecycle ownership
+
+Partial application ownership means that application development is done outside of Giant Swarm, therefore this is the case of upstream applications that we are deploying to our clusters.
+
+To name a few examples, we run, but do not develop, Prometheus, Grafana, Cluster API controllers, Kyverno, different CNI, CSI, CPI apps, then core Kubernetes components, such as API server, scheduler, controller manager, etcd, kubelet, etc.
+
+Partial application ownership of these components means that a Giant Swarm team owns:
+- Release engineering, fully or partially;
+- Operations, fully or partially, depending if the team has sole or shared ownership of the application, which is discussed in the next section.
+
+**Release engineering is owned fully** in case of forked projects, where Giant Swarm team builds the application from the forked repository, then tests, packages and finally delivers it.
+
+**Release engineering is owned partially** when Giant Swarm team does not build the app, but, for example, re-tagged container image is used. Here we have two cases:
+- Giant Swarm team is packaging app as a Helm chart in the app repository, by building required Helm templates for deploying re-tagged container image and optionally using existing upstream Kubernetes manifests instead of writing Helm templates from scratch.
+- Giant Swarm is repackaging or importing existing upstream Helm charts into Giant Swarm application repositories.
```

Comment:
```suggestion
- Giant Swarm team is packaging the app as a Helm chart in the app repository, by building required Helm templates for deploying re-tagged container images and optionally using existing upstream Kubernetes manifests instead of writing Helm templates from scratch.
- Giant Swarm is repackaging or importing existing upstream Helm charts into Giant Swarm application repositories.
```